ui/src/wallet.ts

```diff
@@ -53,10 +46,42 @@ export interface Wallet extends svelteStore.Readable<State> {
   disconnect(): Promise<void>;
   provider: ethers.providers.Provider;
   signer: ethers.Signer;
-  account(): Account | undefined;
+  // Returns the address of the wallet account if the wallet is
+  // connected.
+  getAddress(): string | undefined;
   destroy(): void;
 }
 
+export const accountBalancesStore = svelteStore.writable<{
+  dai: ethers.BigNumber | null;
+  eth: ethers.BigNumber | null;
+}>({ dai: null, eth: null });
```

Comment:
You’re right. Ideally they would be separate. But updating these values also requires some common logic (the mutex to prevent race conditions) which we would need to extract. I’ll leave this as a follow-up.